### PR TITLE
Try/trying edit

### PR DESCRIPTION
**Pull request by de-b**
Do not merge this with any branch, on this branch just trying to edit the todo task, but I am not getting any idea, how to do it.

### CODE REVIEW

File: src/components/ListOfTodos.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import ToDoItem from "./ToDoItem";
 import Button from "@material-ui/core/Button";
 
 function ListOfTodos(props) {
-  const { todos, clearItems, deleteItem } = props;
+  const { todos, clearItems, deleteItem, editItem } = props;
```
Comment (ayrbox) [inline src/components/ListOfTodos.js]: Best practice is to prefix event with "on" - `onDeleteItem`  and `onEditItem`

File: src/components/ToDo.js [hunk 2]
```diff
@@ -43,20 +49,37 @@ const ToDo = () => {
...
+  const deleteItem = (id) => {
+    let temp = todos.filter((item) => item.id !== id);
     setTodos(temp);
+    handleAlert({ type: "danger", text: "item deleted" });
+  };
+
+  const editItem = (id) => {
+    //console.log(id);
+    const singleItem = todos.find((item) => item.id === id);
+    console.log(singleItem);
```
Comment (ayrbox) [inline src/components/ToDo.js]: Function name should be`handleEditItem` and it does nothing at all. Its incomplete.

**Review by ayrbox — CHANGES_REQUESTED** [review]
# Few Issues

1. Better to branch off of master usually.  If the branch depends other feature branch it the base shoudl point the branch from which its branched off.  the diff has same thing twice in both PRs.

2. prefix props with `on` for any event props and `handle` for the functions that handles the events. Just abest practice.

3. Edit Item is incomplete.

File: src/components/ToDo.js (hunk 2 above)
Comment (de-b) [reply src/components/ToDo.js]: Malai edit garnai aayena, I am checking a tutorial and will fix this

**Comment by de-b** [comment]
I have fixed the edit on todo section.

File: src/components/ToDo.js [hunk 3]
```diff
@@ -25,38 +26,78 @@ const mainTodos = [
 const ToDo = () => {
   const [todos, setTodos] = useState(mainTodos);
   const [inputText, setInputText] = useState("");
+  const [alert, setAlert] = useState({ show: false });
+  const [editingItem, setEditingItem] = useState(false);
+  const [id, setId] = useState(0);
```
Comment (ayrbox) [inline src/components/ToDo.js:31]: `editingItem` and `id` 
This state is duplicate. One of them should be enough for editing.

File: src/components/ToDo.js [hunk 4]
```diff
@@ -25,38 +26,78 @@ const mainTodos = [
...
     setTodos(temp);
+    handleAlert({ type: "danger", text: "item deleted" });
+    setInputText("");
+  };
+
+  const handleEdit = (id) => {
+    //console.log(id);
+    const singleItem = todos.find((item) => item.id === id);
+    setEditingItem(true);
+    setId(id);
```
Comment (ayrbox) [inline src/components/ToDo.js:92]: Same here.

File: src/components/ToDoForm.js [hunk 5]
```diff
@@ -11,18 +11,21 @@ const useStyles = makeStyles((theme) => ({
...
       <TextField
         className={classes.fullWidth}
         id="standard-basic"
         label="Add Todo item"
         value={inputText}
         onChange={handleChange}
       />
-      <button>Submit</button>
+      {/* <Button variant="outlined">Submit</Button> */}
+      <button className="btn">{editingItem ? "Edit" : "Submit"}</button>
```
Comment (ayrbox) [inline src/components/ToDoForm.js:28]: Please remove commented code.

File: src/components/ToDo.js [hunk 6]
```diff
@@ -65,11 +106,14 @@ const ToDo = () => {
             inputText={inputText}
             handleChange={handleChange}
             handleSubmit={handleSubmit}
+            editingItem={editingItem}
+            handleEdit={handleEdit}
```
Comment (ayrbox) [inline src/components/ToDo.js:110]: Event props definition should be prefix with `on`
```suggestion
            onEdit={handleEdit}
```

File: src/components/ToDo.js [hunk 7]
```diff
@@ -25,38 +26,78 @@ const mainTodos = [
 const ToDo = () => {
   const [todos, setTodos] = useState(mainTodos);
   const [inputText, setInputText] = useState("");
+  const [alert, setAlert] = useState({ show: false });
```
Comment (ayrbox) [inline src/components/ToDo.js:29]:
```suggestion
  const [showAlert, setShowAlert] = useState(false);
```
It should be simpler. No need to map object.

File: src/components/ToDo.js [hunk 8]
```diff
@@ -65,11 +106,14 @@ const ToDo = () => {
...
             handleChange={handleChange}
             handleSubmit={handleSubmit}
+            editingItem={editingItem}
+            handleEdit={handleEdit}
           />
           <ListOfTodos
             todos={todos}
             clearItems={clearItems}
             deleteItem={deleteItem}
+            handleEdit={handleEdit}
```
Comment (ayrbox) [inline src/components/ToDo.js:116]:
```suggestion
            onEdit={handleEdit}
```

File: src/components/ToDoForm.js [hunk 9]
```diff
@@ -11,18 +11,21 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-function ToDoForm({ inputText, handleChange, handleSubmit }) {
+function ToDoForm({ inputText, handleChange, handleSubmit, editingItem }) {
```
Comment (ayrbox) [inline src/components/ToDoForm.js:14]:
```suggestion
function ToDoForm({ inputText, handleChange, handleSubmit, itemToEdit }) {
```
`itemToEdit` makes more sense when reading code.

**Review by ayrbox — CHANGES_REQUESTED** [review]

**Pull request merged**